### PR TITLE
fix: wds components

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Fixed WDS components bugs

Fixes #
#33077
#33078
#32689
#32919

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8926075231>
> Commit: 0981cf8a45151cb3dc24b938efb7d79c7fefc766
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8926075231&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by ramsaptami** [comment]
everything here looks here @KelvinOm 

just got one feedback wrt, #32689 the fix is better but not perfect - the gap is still noticeable
image

File: app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx [hunk 1]
```diff
@@ -57,8 +57,10 @@ const _TextInput = (props: TextInputProps, ref: HeadlessTextInputRef) => {
       return (
         <IconButton
           color="neutral"
+          excludeFromTabOrder
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx:60]: What does this prop do?🤔

**Review by jsartisan — DISMISSED** [review]

File: app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx:60]: add tabIndex="-1" so button will not access via tabulation

**Comment by KelvinOm** [comment]
> everything here looks here @KelvinOm
> 
> just got one feedback wrt, #32689 the fix is better but not perfect - the gap is still noticeable image

@ramsaptami  Added fix for this, should be better now.

**Review by ichik — APPROVED** [review]

**Review by ramsaptami — APPROVED** [review]

**Pull request merged**